File: Java/src/Iniciante/Solucao_1005.java

import java.io.IOException;
import java.util.Scanner;

public class Solucao_1005 {
    public static boolean CalcularSeEstaEntre0E10(double num){// Metodo estatico para num esta entre 0 e 10
        if(num >= 0 && num <= 10){ //if olha se u num entra entre 0 e 10
            return true; //se sim retorna um true
        }else {
            return false;//senão retorna um false
        }
    }

    public static void  main(String[] args)throws IOException {
        var input = new Scanner(System.in);//objeto Scanner para input


        while(true) {// loop infinito
            double pesoUm = 3.5;//peso de A
            double pesoDois = 7.5; //peso de B
            double a = input.nextFloat(); //leitura de A e B
            double b = input.nextFloat();

            boolean aValido = CalcularSeEstaEntre0E10(a);//aqui eu chamo o metodo que se estive entre 0 e 10 aValido receber um TRUE
            boolean bValido = CalcularSeEstaEntre0E10(b);//caso não receber um FALSE esse metodo nas duas variaveis (A e B)

            if(aValido && bValido){// Se 'a' e 'b' forem true vai imprimir o resultado
                double resultando = (a * pesoUm + b * pesoDois)/(pesoUm + pesoDois); //formula da media
                System.out.printf("MEDIA = %.5f%n",resultando);//impressão com cinco casas decimais
                break;// se estou no bloco do if então aqui para o fluxo do while
            }//se não o loop vai roda novamente
        }



        input.close();// fechamento do input
    }
}

File: Python/Iniciante/Solucao_1005.py
try:

    def entre_0e10(num):#função para verificar ser o num esta entre 0 e 10
        if num >= 0 and num <= 10:
            return True# se esta entre 0 e 10 retorna true
        else:
            return False #senão false

    while True:#while infinito
        a = float(input()) #leitura de A e B
        b = float(input())
        a_valido = entre_0e10(a) # a_valido e b_valido chama a função entre_0e10 para ver se A e B estão entre 0 e 10
        b_valido = entre_0e10(b)
        peso_um = 3.5# peso de A e B
        peso_dois = 7.5

        if a_valido and b_valido:# Se apos a função A e B retornam true então
            resulto = (a * peso_um + b * peso_dois) / (peso_um + peso_dois)#resultado receber a media
            print(f"MEDIA = {resulto:.5f}")# print com 5 casas decimais
            break# aqui se a codição do if e verdadeira então trava o fluxo e para o while
except (ValueError, TypeError):
    exit()